docs/06-concepts/18-testing/02-the-basics.md

```diff
@@ -0,0 +1,248 @@
+
+|Property|Type|Default|Description|
+|:-----|:-----|:---:|:-----|
+|`authentication`|`AuthenticationOverride?`|`AuthenticationOverride.unauthenticated()`|See section [Setting authenticated state](#setting-authenticated-state).|
+|`enableLogging`|`bool?`|`false`|Wether logging is turned on for the session.|
+
+The `copyWith` method creates a new unique session builder with the provided properties. This can then be used in endpoint calls (see section [Setting authenticated state](#setting-authenticated-state) for an example).
+
+To build out a `Session` (to use for [database calls](#seeding-the-database) or [pass on to functions](advanced-examples##business-logic-depends-on-session)), simply call the `build` method:
+
+```dart
+Session session = sessionBuilder.build();
+```
+
+Given the properties set on the session builder through the `copyWith` method, this returns a Serverpod `Session` that has the corresponding state.
+
+### Setting authenticated state {#setting-authenticated-state}
+
+To control the authenticated state of the session, the `AuthenticationOverride` class can be used.
+
+To create an unauthenticated override (this is the default value for new sessions), call `AuthenticationOverride unauthenticated()`:
+
+```dart
+static AuthenticationOverride unauthenticated();
+```
+
+To create an authenticated override, call `AuthenticationOverride.authenticationInfo(...)`:
+
+```dart
+static AuthenticationOverride authenticationInfo(
+  int userId,
+  Set<Scope> scopes, {
+  String? authId,
+})
+```
+
+Pass these to `sessionBuilder.copyWith` to simulate different scenarios. Below follows an example for each case:
+
+```dart
+withServerpod('Given AuthenticatedExample endpoint', (sessionBuilder, endpoints) {
+  // Corresponds to an actual user id
+  const int userId = 1234;
+
+  group('when authenticated', () {
+    var authenticatedSessionBuilder = sessionBuilder.copyWith(
+      authentication:
+          AuthenticationOverride.authenticationInfo(userId, {Scope('user')}),
+    );
+
+    test('then calling `hello` should return greeting', () async {
+      final greeting = await endpoints.authenticatedExample
+          .hello(authenticatedSessionBuilder, 'Michael');
+      expect(greeting, 'Hello, Michael!');
+    });
+  });
+
+  group('when unauthenticated', () {
+    var unauthenticatedSessionBuilder = sessionBuilder.copyWith(
+      authentication: AuthenticationOverride.unauthenticated(),
+    );
+
+    test(
+        'then calling `hello` should throw `ServerpodUnauthenticatedException`',
+        () async {
+      final future = endpoints.authenticatedExample
+          .hello(unauthenticatedSessionBuilder, 'Michael');
+      await expectLater(
+          future, throwsA(isA<ServerpodUnauthenticatedException>()));
+    });
+  });
+});
+```
+
+### Seeding the database {#seeding-the-database}
```

Comment:
Question: Is the additional `{#seeding-the-database}` really required here?

Reply:
As long as there are no collisions I think this name is automatically created.

If it is not required, please check this for all headers that have this addition.

Reply:
Oh, I didn't realize it was auto-generated. Will check all of these 👍